pkg/cloudprovider/aws/instance.go

```diff
@@ -249,17 +249,26 @@ func (p *InstanceProvider) instanceToNode(ctx context.Context, instance *ec2.Ins
 			if injection.GetOptions(ctx).GetAWSNodeNameConvention() == options.ResourceName {
 				nodeName = aws.StringValue(instance.InstanceId)
 			}
+			zone := aws.StringValue(instance.Placement.AvailabilityZone)
+			region := ""
+			for _, offering := range instanceType.Offerings() {
+				if offering.Zone == zone {
+					region = offering.Region
+					break
+				}
+			}
 			return &v1.Node{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: nodeName,
 					Labels: map[string]string{
-						v1.LabelTopologyZone:       aws.StringValue(instance.Placement.AvailabilityZone),
+						v1.LabelTopologyZone:       zone,
+						v1.LabelTopologyRegion:     region,
```

Comment:
I'd happily skip applying this label. Not sure I see the value -- the kubelet will put it on, anyways.